core/modules/ccontrol/MergingHandler.cc

```diff
@@ -76,93 +78,84 @@ MergingHandler::~MergingHandler() {
 
 const char* MergingHandler::getStateStr(MsgState const& state) {
     switch(state) {
-    case MsgState::INVALID:          return "INVALID";
-    case MsgState::HEADER_SIZE_WAIT: return "HEADER_SIZE_WAIT";
-    case MsgState::RESULT_WAIT:      return "RESULT_WAIT";
-    case MsgState::RESULT_RECV:      return "RESULT_RECV";
-    case MsgState::RESULT_EXTRA:     return "RESULT_EXTRA";
-    case MsgState::HEADER_ERR:       return "HEADER_ERR";
-    case MsgState::RESULT_ERR:       return "RESULT_ERR";
+    case MsgState::HEADER_WAIT: return "HEADER_WAIT";
+    case MsgState::RESULT_WAIT: return "RESULT_WAIT";
+    case MsgState::RESULT_RECV: return "RESULT_RECV";
+    case MsgState::HEADER_ERR:  return "HEADER_ERR";
+    case MsgState::RESULT_ERR:  return "RESULT_ERR";
     }
     return "unknown";
 }
 
-bool MergingHandler::flush(int bLen, BufPtr const& bufPtr, bool& last, bool& largeResult, int& nextBufSize) {
+
+bool MergingHandler::flush(int bLen, BufPtr const& bufPtr, bool& last, bool& largeResult,
+                           int& nextBufSize) {
     LOGS(_log, LOG_LVL_DEBUG, "From:" << _wName << " flush state="
          << getStateStr(_state) << " blen=" << bLen << " last=" << last);
-    if ((bLen < 0) || (bLen != (int)bufPtr->size())) {
-        if (_state != MsgState::RESULT_EXTRA) {
-            LOGS(_log, LOG_LVL_ERROR, "MergingRequester size mismatch: expected " <<
-                 bufPtr->size() << " got " << bLen);
-            // Worker sent corrupted data, or there is some other error.
-        }
+
+    if (bLen < 0) {
```

Comment:
Can blen be zero? Is that a valid state?

Reply:
Yes, 0 indicates that there's nothing following this header and the message is done.